libs-scala/scala-utils/src/test/scala/com/daml/scalautil/future/FutureConversionSpec.scala

```diff
@@ -0,0 +1,44 @@
+package com.daml.scalautil.future
+
+import java.util.concurrent.{CompletableFuture, CompletionStage}
+
+import com.daml.scalautil.future.FutureConversionSpec.TestException
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AsyncWordSpec
+
+import scala.concurrent.Future
+import scala.jdk.FutureConverters._
+
+class FutureConversionSpec extends AsyncWordSpec with Matchers {
```

Comment:
AFAIU, we have 2 test cases that produce the `CompletionException` in 2 different ways. I would add a test case for a failure that is not wrapped by `CompletionException` and one for a successful case.

Reply:
Good point, added the test cases you mentioned.